Source/Core/Core/Analytics.cpp

```diff
@@ -270,7 +270,8 @@ void DolphinAnalytics::MakeBaseBuilder()
   builder.AddData("android-version", s_get_val_func("DEVICE_OS"));
 #elif defined(__APPLE__)
   builder.AddData("os-type", "osx");
-
+//objc_msgSend is only available on x86
+#ifndef _M_ARM_64
   // id processInfo = [NSProcessInfo processInfo]
   id processInfo = reinterpret_cast<id (*)(Class, SEL)>(objc_msgSend)(
```

Comment:
Under arm64, objc_msgSend_stret does not exist separately for struct-related results, but objc_msgSend itself does and can be relied on. While possibly not the cleanest, my solution was to do something along the following:

```c
#if _M_ARM_64
#define msgSend objc_msgSend
#else
#define msgSend objc_msgSend_stret
#endif

// NSOperatingSystemVersion version = [processInfo operatingSystemVersion]
OSVersion version = reinterpret_cast<OSVersion (*)(id, SEL)>(msgSend)(
```

Reply:
Awesome! This is a much better way to handle the analytics. I will incorporate this in the next push.